akka-persistence/src/test/scala/akka/persistence/fsm/PersistentFSMSpec.scala

```diff
@@ -329,6 +329,20 @@ abstract class PersistentFSMSpec(config: Config) extends PersistenceSpec(config)
       persistentEventsStreamer ! PoisonPill
       expectTerminated(persistentEventsStreamer)
     }
+
+    "allow cancelling stateTimeout by issuing forMax(Duration.Inf)" in {
+      val p = TestProbe()(system)
+
+      val fsm = system.actorOf(TimeoutFSM.props(p.ref))
+
+      p.expectMsg(PersistentFSM.StateTimeout)
+
+      fsm ! TimeoutFSM.OverrideTimeoutToInf
+      p.expectMsg(TimeoutFSM.OverrideTimeoutToInf)
+      p.expectNoMsg(3.seconds)
```

Comment:
1.second would be enough if the other timeout is shorter (given that we run the tests many times)